pkg/frontend/querymiddleware/codec.go

```diff
@@ -353,15 +353,20 @@ func (c prometheusCodec) EncodeRequest(ctx context.Context, r Request) (*http.Re
 }
 
 func (c prometheusCodec) DecodeResponse(ctx context.Context, r *http.Response, _ Request, logger log.Logger) (Response, error) {
-	if r.StatusCode/100 == 5 {
-		return nil, httpgrpc.ErrorFromHTTPResponse(&httpgrpc.HTTPResponse{
-			Code: int32(r.StatusCode),
-			Body: mustReadResponseBody(r),
-		})
-	} else if r.StatusCode == http.StatusTooManyRequests {
+	switch r.StatusCode {
+	case http.StatusServiceUnavailable:
```

Comment:
Could you please add a unit test for this in codec_test.go?

Reply:
Done, PTAL.

Reply:
Thank you